modules/iam-read-only-policy/variables.tf

```
@@ -27,6 +27,12 @@ variable "allowed_services" {
   type        = list(string)
 }
 
+variable "denied_actions" {
+  description = "List of actions to deny by default"
+  type        = list(string)
+  default     = ["s3:GetObject"]
```

```suggestion
  default     = []
```

It should be up to the user to decide what should be denied. Actually, `s3:GetObject` is pure read-only action :)

---

It is pure read indeed
This default is more of a guardrail since s3:GetObject gives to all resources will unintentionally give access to sensitive data. If everyone would use CMK's for the sensitive data according to security best practices then s3:GetObject would be not a problem because you would also need km:Decrypt for the key to get the object.

Anyhow, I see your point but got many people to point out that giving s3:GetObject is dangerous. So user's preference here is to disable it or replace it with more stuff.

Perhaps I should add this explanation to the description. What do you think?